src/Rule/Required.php

```diff
@@ -41,10 +41,11 @@ public function __construct(
         ?callable $emptyCallback = null,
         private bool $skipOnError = false,
         /**
-         * @psalm-var Closure(mixed, ValidationContext):bool|null
+         * @var Closure(mixed, ValidationContext):bool|null
          */
         private ?Closure $when = null,
     ) {
+        /** @psalm-suppress MixedPropertyTypeCoercion */
```

Comment:
It's not need. Fix annotations for $emptyCallback:

```php
    /**
     * @param callable $emptyCallback
     *
     * @psalm-param EmptyCallback $emptyCallback
     */
```

Reply:
Missed that it's not property, but parameter here. Added with some adjustments:

- `@param callable $emptyCallback` is redundant because it has the same type hint and no additional description.
- Unlike according property, the parameter can be nullable.